src/implementations/eigh.jl

```diff
@@ -85,6 +115,25 @@ function eigh_trunc!(A::AbstractMatrix, DV, alg::TruncatedAlgorithm)
     return truncate!(eigh_trunc!, (D, V), alg.trunc)
 end
 
+# Diagonal logic
+# --------------
+function eigh_full!(A::Diagonal, DV, alg::DiagonalAlgorithm)
+    check_input(eigh_full!, A, DV, alg)
+    D, V = DV
+    D === A || (diagview(D) .= real.(diagview(A)))
```

Comment:
Do we simply discard possible imaginary values on the diagonal? I guess we also don't check for hermiticity in the general matrix case. Do we want to add such a check in `check_input`?

Reply:
I just double-checked and noticed that the LAPACK implementations are actually checking for symmetric/hermitian, so I'll adapt this here as well.